gravitee-am-repository/gravitee-am-repository-mongodb/src/main/java/io/gravitee/am/repository/mongodb/management/upgrader/ReporterReferenceMongoUpgrader.java

```diff
@@ -0,0 +1,57 @@
+/**
+ * Copyright (C) 2015 The Gravitee team (http://gravitee.io)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.gravitee.am.repository.mongodb.management.upgrader;
+
+import com.mongodb.reactivestreams.client.MongoDatabase;
+import io.gravitee.am.model.ReferenceType;
+import io.gravitee.am.repository.mongodb.management.internal.model.ReporterMongo;
+import io.gravitee.node.api.upgrader.Upgrader;
+import io.reactivex.rxjava3.core.Single;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Component;
+
+import static com.mongodb.client.model.Filters.exists;
+import static com.mongodb.client.model.Updates.combine;
+import static com.mongodb.client.model.Updates.rename;
+import static com.mongodb.client.model.Updates.set;
+
+@RequiredArgsConstructor
+@Slf4j
+@Component
+public class ReporterReferenceMongoUpgrader implements Upgrader {
+    private final MongoDatabase mongo;
+
+    @Override
+    public boolean upgrade() {
+        var reportersCollection = mongo.getCollection("reporters", ReporterMongo.class);
+        return Single.fromPublisher(reportersCollection.updateMany(exists("domainId"),
+                        combine(
+                                set("referenceType", ReferenceType.DOMAIN),
+                                rename("domainId", "referenceId"))
```

Comment:
**issue:** please don't do a rename, if you are renaming the field then a Gateway in previous version will not be able to start the reporter (or even just not be able to start at all) because the domain field will be missing. We have to guaranty some compatibility accross version. In 4.6 or later when the domain attribute will removed from the ReporterMongo class, we will have to implement a upgrader to clean up the collection (same for liquibase on RDBMS)